addons/mail/models/mail_activity.py

```diff
@@ -577,6 +577,17 @@ def action_open_document(self):
             'view_mode': 'form',
         }
 
+    def action_snooze(self):
```

Comment:
In `addons/crm/models/crm_lead.py`, we also have an `action_snooze` with the same logic: In that method, we retrieve the next activity that is linked to the current lead and the current user and we "snooze" it.

Couldn't we avoid code duplication and call from this action the `action_snooze` function we add here?

So, in `addons/crm/models/crm_lead.py`:

```python
def action_snooze(self):
    self.ensure_one()
    my_next_activity = self.activity_ids.filtered(lambda activity: activity.user_id == self.env.user)[:1]
    my_next_activity.action_snooze()
    return True
```

Note: In the `crm.lead` model, `activity_ids` is a M2O field of type `mail.activity` that is shipped with the mail activity mixin (see: `MailActivityMixin`).